src/App.tsx

```diff
@@ -387,14 +379,12 @@ const App = (): JSX.Element => {
             <React.Suspense fallback={<FullLoadingSpinner />}>
               <Switch location={location}>
                 <Route exact path={PAGES.VAULTS}>
-                  <Vaults />
+                  {bridgeLoaded ? <Vaults /> : <FullLoadingSpinner />}
```

Comment:
In `Vaults` and `Vault`, the bridge needs to be loaded because there's a function that requires that. We could properly handle it but it requires some sophisticated fallback structure so I'd suggest handling that after this PR if we really want to do it.

Reply:
@peterslany cc @tomjeatt 
Yes, indeed. I think we should do like so according to best practices but there's a few things we need to consider before that. @nud3l and @alexeiZamyatin wanted the UI to be visible as much as possible even when the bridge is not loaded. For example, let's assume that the bridge is not loaded for some reason. In that case, we would not render any UI but just a description or something to let users know that there's a problem with the bridge. As far as I know, this is not what we want in the UI workflow. I think we should brainstorm it but I agree with you. (totally :))

Let me create a ticket so that we can pick this issue up down the line.